src/shared/redux/slices/addPublicationSlice.ts

```diff
@@ -0,0 +1,87 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from '../rootReducer';
+import { Collaborator } from './loadPublicationsSlice';
+
+export type NewPublication = {
+  // for now there is no image to store, to be added later
+  imagePath?: string;
+  publicationName: string;
+  description: string;
+  collaborators: Collaborator[];
+  packageManager: string;
+  useSass: boolean;
+  useTypescript: boolean;
+  step: number;
+};
+
+// type created to avoid an issue with both unclear modification type
+// and index access writes on union of keys, which would complicate the
+// code significantly more
+type PublicationModification =
+  | {
+      field: keyof Omit<
+        NewPublication,
+        'useTypescript' | 'useSass' | 'collaborators' | 'step'
+      >;
+      value: string;
+    }
+  | {
+      field: keyof Pick<NewPublication, 'useTypescript' | 'useSass'>;
+      value: boolean;
+    }
+  | {
+      field: keyof Pick<NewPublication, 'step'>;
+      value: number;
+    };
```

Comment:
How about structuring this type that way? 😄 I feel it's a bit more readable and takes up less space 😆 
```suggestion
type PublicationModification = {
  value: string | boolean | number;
  field: keyof (
    | Pick<NewPublication, 'useTypescript' | 'useSass'>
    | Pick<NewPublication, 'step'>
    | Omit<
        NewPublication,
        'useTypescript' | 'useSass' | 'collaborators' | 'step'
      >
  );
};
```

Reply:
@basia-borkowska it takes up less space but it does not have the same semantic meaning and therefore, does not unfortunately allow to achieve the correct typings. This typings are for matching the `value` type with a specific `field` type. In your suggestion the user can set a value of `boolean` to any field for example, which is not desired. Whereas in the original typings if the `value` is `boolean` the field can only be set to `useTypescript` or `useSass`.

@JKusiak has more insight on the topic if you'd like to inquire further.